Add information about RHEL 8 to RHEL 9 upgrade (#1756)
theforeman/foreman-documentation: guides/common/modules/proc_upgrading-hosts-to-next-major-rhel-release.adoc

[id="Upgrading_Hosts_to_Next_Major RHEL_release_{context}"]
= Upgrading Hosts to Next Major {RHEL} Release

You can use a job template to upgrade your {RHEL} hosts to the next major version.
Below upgrade paths are possible:

* {RHEL} 7 to {RHEL} 8
* {RHEL} 8 to {RHEL} 9

> .Prerequisites

Reviewer: ```suggestion

.Prerequisites
```

* Ensure that your {RHEL} hosts meet the requirements for the upgrade.
ifndef::orcharhino[]

> ** For {RHEL} 7 to {RHEL} 8 upgrade, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/upgrading_from_rhel_7_to_rhel_8/planning-an-upgrade_upgrading-from-rhel-7-to-rhel-8[Planning an upgrade] in the _Upgrading from RHEL 7 to RHEL 8_ guide.

Reviewer: ```suggestion
** For {RHEL} 7 to {RHEL} 8 upgrade, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/upgrading_from_rhel_7_to_rhel_8/planning-an-upgrade_upgrading-from-rhel-7-to-rhel-8[Planning an upgrade] in _Upgrading from RHEL 7 to RHEL 8_.
```

I feel like this would be match the current guide title usage. My suggestion, if applied, would also affect all other links similarly.

endif::[]
* Enable remote execution feature on {Project}.
For more information, see xref:Configuring_and_Setting_Up_Remote_Jobs_{context}[].
* Distribute {Project} SSH keys to the hosts that you want to upgrade.
For more information, see xref:Distributing_SSH_Keys_for_Remote_Execution_{context}[].

.Procedure
. On {Project}, enable the Leapp plugin:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-plugin-leapp
----
. In the {ProjectWebUI}, navigate to *Hosts* > *All Hosts*.
. Select the hosts that you want to upgrade to the next major {RHEL} version.
. In the upper right of the Hosts window, from the *Select Action* list, select *Preupgrade check with Leapp*.
. Click *Submit* to start the pre-upgrade check.
. When the check is finished, click the *Leapp preupgrade report* tab to see if Leapp has found any issues on your hosts.
Issues that have the *Inhibitor* flag are considered crucial and are likely to break the upgrade procedure.

> Some issues might have documentation linked that describe how to fix them.

Reviewer 2: ```suggestion
Some issues might have documentation linked that describes how to fix them.
```

> . Optional: If you have issues that have commands associated with them, you can fix them with a remote job.

Reviewer: maybe "using remote execution" instead of "with"?

To do that, select these issues, click the *Fix Selected* button, and submit the job.

> . Once the issues are fixed, click the *Rerun* button, and then click *Submit* to run the pre-upgrade check again to verify that the {RHEL} hosts you are upgrading do not have any issues and are ready to be upgraded.

Reviewer: ```suggestion
. After the issues are fixed, click the *Rerun* button, and then click *Submit* to run the pre-upgrade check again to verify that the hosts you are upgrading do not have any issues and are ready to be upgraded.
```

> . When your systems are ready for the upgrade, click the *Run Upgrade* button and click *Submit* to start the upgrade.

Reviewer: ```suggestion
. When your hosts are ready for the upgrade, click the *Run Upgrade* button and click *Submit* to start the upgrade.
```

How do I know when they're ready? Do you mean "After you have solved all issues on your hosts, ..."?

> [id="Upgrading_Hosts_to_Next_Major RHEL_release_{context}"]

Reviewer: ```suggestion
[id="Upgrading_Hosts_to_Next_Major_RHEL_Release_{context}"]
```

Reviewer: @AkshayGadhaveRH Anchors cannot contain whitespace.

= Upgrading Hosts to Next Major {RHEL} Release

> You can use a job template to upgrade your {RHEL} hosts to the next major version.

Reviewer: I'm thinking either also use "major release", or reword the title, anchor, file name, and include statement. I'm slightly leaning towards "major version".

Author: Hey @maximiliankolb just to confirm:
1. Do you mean changing the title to Upgrading Hosts to Next Major {RHEL} Version?
2. Do you want to drop {RHEL} from the title as well?
3. Replace `release` with `version` in the anchor and filename too.

Author: Gotcha! I have replaced "version" with "release".

[id="Upgrading_Hosts_to_Next_Major RHEL_release_{context}"]

> = Upgrading Hosts to Next Major {RHEL} Release

Reviewer: ```suggestion
= Upgrading Hosts to The Next Major {RHEL} Release
```

I am unsure about the wording here. cc @Levi-Leah

Reviewer 3: > I am unsure about the wording here.

imho the wording is fine, I left a comment on prerequisites tho

You can use a job template to upgrade your {RHEL} hosts to the next major version.
Below upgrade paths are possible:

> * Red Hat Enterprise Linux 7 to Red Hat Enterprise Linux 8

Reviewer: ```suggestion
* {RHEL} 7 to {RHEL} 8
```

> * Red Hat Enterprise Linux 8 to Red Hat Enterprise Linux 9

Reviewer: ```suggestion
* {RHEL} 8 to {RHEL} 9
```

> .Prerequisites

Reviewer 3: I'd do smth like this for the prerequisites to avoid nesting:

```
.Prerequisites
* Ensure that your {RHEL} hosts meet the requirements for the upgrade.
ifndef::orcharhino[]
** For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/upgrading_from_rhel_7_to_rhel_8/planning-an-upgrade_upgrading-from-rhel-7-to-rhel-8[Planning an upgrade] in the _Upgrading from RHEL 7 to RHEL 8_ guide.
** For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/upgrading_from_rhel_8_to_rhel_9/planning-an-upgrade_upgrading-from-rhel-8-to-rhel-9[Planning an upgrade] in the _Upgrading from RHEL 8 to RHEL 9_ guide.
endif::[]
* Prepare your hosts for the upgrade.
ifndef::orcharhino[]
** For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/upgrading_from_rhel_7_to_rhel_8/index#preparing-a-rhel-7-system-for-the-upgrade_upgrading-from-rhel-7-to-rhel-8[Preparing a RHEL 7 system for the upgrade].
** For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/upgrading_from_rhel_8_to_rhel_9/assembly_preparing-for-the-upgrade_upgrading-from-rhel-8-to-rhel-9#preparing-a-rhel-8-system-for-the-upgrade_upgrading-from-rhel-8-to-rhel-9[Preparing a RHEL 8 system for the upgrade].
endif::[]
* Enable remote execution feature on {Project}.
For more information, see xref:Configuring_and_Setting_Up_Remote_Jobs_{context}[].
* Distribute {Project} SSH keys to the hosts that you want to upgrade.
For more information, see xref:Distributing_SSH_Keys_for_Remote_Execution_{context}[].
```

> * For upgrading {RHEL} 7 hosts to {RHEL} 8

Reviewer: ```suggestion
* For upgrading {RHEL} 7 hosts to {RHEL} 8:
```

** Ensure that your {RHEL} 7 hosts meet the requirements for the upgrade to {RHEL} 8.
ifndef::orcharhino[]
For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/upgrading_from_rhel_7_to_rhel_8/planning-an-upgrade_upgrading-from-rhel-7-to-rhel-8[Planning an upgrade] in the _Upgrading from RHEL 7 to RHEL 8_ guide.
endif::[]
** Prepare your hosts for the upgrade.
ifndef::orcharhino[]
For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/upgrading_from_rhel_7_to_rhel_8/index#preparing-a-rhel-7-system-for-the-upgrade_upgrading-from-rhel-7-to-rhel-8[Preparing a RHEL 7 system for the upgrade] in the _Upgrading from RHEL 7 to RHEL 8_ guide.
endif::[]

> * For upgrading {RHEL} 8 hosts to {RHEL} 9

Reviewer: ```suggestion
* For upgrading {RHEL} 8 hosts to {RHEL} 9:
```

** Ensure that your {RHEL} 8 hosts meet the requirements for the upgrade to {RHEL} 9.
ifndef::orcharhino[]
For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/upgrading_from_rhel_8_to_rhel_9/planning-an-upgrade_upgrading-from-rhel-8-to-rhel-9[Planning an upgrade] in the _Upgrading from RHEL 8 to RHEL 9_ guide.
endif::[]

> ** Prepare your hosts for the upgrade.
> ifndef::orcharhino[]
> For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/upgrading_from_rhel_8_to_rhel_9/assembly_preparing-for-the-upgrade_upgrading-from-rhel-8-to-rhel-9#preparing-a-rhel-8-system-for-the-upgrade_upgrading-from-rhel-8-to-rhel-9[Preparing a RHEL 8 system for the upgrade] in the _Upgrading from RHEL 8 to RHEL 9_ guide.
> endif::[]

Reviewer: ```suggestion
```

This seems like a duplication.

Author: Hey @maximiliankolb I have restructured the doc a bit as @Levi-Leah suggested. Looks like this was addressed in the restructuring. Should I mark this as resolved too?

# {foreman-installer} --enable-foreman-plugin-leapp
----
. In the {ProjectWebUI}, navigate to *Hosts* > *All Hosts*.
. Select the hosts that you want to upgrade to the next major {RHEL} version.
. In the upper right of the Hosts window, from the *Select Action* list, select *Preupgrade check with Leapp*.
. Click *Submit* to start the pre-upgrade check.

> . When the check is finished, click the *Leapp preupgrade report* tab to see if Leapp has found any issues on the hosts.

Reviewer: ```suggestion
. When the check is finished, click the *Leapp preupgrade report* tab to see if Leapp has found any issues on your hosts.
```

Issues that have the *Inhibitor* flag are considered crucial and are likely to break the upgrade procedure.
Some issues might have documentation linked that describe how to fix them.
. Optional: If you have issues that have commands associated with them, you can fix them with a remote job.
To do that, select these issues, click the *Fix Selected* button, and submit the job.

> . After you fixed the issues, click the *Rerun* button, and then click *Submit* to run the pre-upgrade check again to verify that the {RHEL} hosts you are upgrading do not have any issues and are ready to be upgraded.

Reviewer 2: ```suggestion
. Once the issues are fixed, click the *Rerun* button, and then click *Submit* to run the pre-upgrade check again to verify that the {RHEL} hosts you are upgrading do not have any issues and are ready to be upgraded.
```
